ios/RNTPTDocumentView.m

```diff
@@ -2052,6 +2063,36 @@ -(NSString*)generateXfdfCommand:(PTVectorAnnot*)added modified:(PTVectorAnnot*)m
     return fdfCommand;
 }
 
+#pragma mark - PTBookmarkViewControllerDelegate
+
+- (void)bookmarkViewController:(PTBookmarkViewController *)bookmarkViewController didModifyBookmark:(PTUserBookmark *)bookmark {
+    [self bookmarksModified];
+}
+
+- (void)bookmarkViewController:(PTBookmarkViewController *)bookmarkViewController didAddBookmark:(PTUserBookmark *)bookmark {
+    [self bookmarksModified];
+}
+
+- (void)bookmarkViewController:(PTBookmarkViewController *)bookmarkViewController didRemoveBookmark:(nonnull PTUserBookmark *)bookmark {
+    [self bookmarksModified];
+}
+
+- (void)bookmarksModified {
+    __block NSString* json;
+    NSError* error;
+    [self.pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+        json = [PTBookmarkManager.defaultManager exportBookmarksFromDoc:doc];
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: There was an error while trying to export the bookmark json on events triggered. %@", error.localizedDescription);
+    }
+    
+    if ([self.delegate respondsToSelector:@selector(bookmarkChanged:bookmarkJson:)]) {
```

Comment:
You could do this check around the whole function. If the person is not listening for changes, we can skip the work of acquiring a lock and extracting the string.